dataset/map_view.cpp

```diff
@@ -170,6 +170,11 @@ Dict<Key, Value> &Dict<Key, Value>::setSlice(const Slice s, const Dict &dict) {
                                    to_string(s.dim()) + "'.");
     }
   }
```

Comment:
Similar to potential issues with data mentioned above: Do we also need to call `Variable::validateSlice` for every mask that neds updating?